controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -253,13 +253,13 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&ocsv1.OCSInitialization{}).
-		Owns(&corev1.Service{}).
-		Owns(&corev1.Secret{}).
-		Owns(&promv1.Prometheus{}).
-		Owns(&corev1.ConfigMap{}).
-		Owns(&promv1.Alertmanager{}).
-		Owns(&promv1.ServiceMonitor{}).
+		For(&ocsv1.OCSInitialization{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.Service{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
service probably has some interested annotations like a serving cert etc for which if removed we want to reconcile, basically I believe we can't add this predicate w/o checking how the dependents are being used in the controller code.

Reply:
removed it.